### DISCUSSION TITLE
[LORAWAN] sending uplink packet… failed code -4

### DISCUSSION THREAD

**Discussion by altinem** [jgromes/RadioLib#977] [General]
Discussion moved from issues #965

**Comment by altinem** [comment]
I just commented the line "node.wipe();" and flashed the board. The OTAA is success and I see "sending uplink packet ....no downlink" message in the serial terminal. When I check the TTN console for activity I dont`t see any activity and uplink message.

**Comment by altinem** [comment]
I just started receiving the message in the TTN console. the message count number 4 and received downlink below.
[LoRaWAN] Sending uplink packet ... received a downlink!
[LoRaWAN] Data:
[LoRaWAN] RSSI: -38.00 dBm
[LoRaWAN] SNR: 10.25 dB
[LoRaWAN] Frequency error: -31.00 Hz

**Comment by altinem** [comment]
I just scheduled a downlink message and this is what I received below:   
[LoRaWAN] Sending uplink packet ... no downlink!
[LoRaWAN] Sending uplink packet ... received a downlink!
[LoRaWAN] Data:         ޭ��
[LoRaWAN] RSSI:         -33.00 dBm
[LoRaWAN] SNR:          9.50 dB
[LoRaWAN] Frequency error:      -31.00 Hz

**Comment by altinem** [comment]
from the TTN website:

Fair Use Policy 
On The Things Network’s The Things Stack Sandbox a Fair Use Policy applies which limits the uplink airtime to 30 seconds per day (24 hours) per node and the downlink messages to 10 messages per day (24 hours) per node. If you use a private network, these limits do not apply, but you still have to be compliant with the governmental and LoRaWAN limits.

**Comment by HeadBoffin** [comment]
You sort of beat me to it - if you wipe the node everything gets reset including the frame counter - you can do that each time or you can add node.saveSession(); after each uplink.

As you have observed, TTN FUP is 10 downlinks a day, the community preference being one a fortnight (automated link checks and LNS MAC commands aside). LW due to potential latency & packet loss is best suited for sending many small sensor readings, not so great for command & control.

**Comment by altinem** [comment]
I have migrated my lorawan projects to Chirpstack after I found out the FUP limitation with TTN. My chirpstack server is already up and running, gateway and end devices already connected and activated. . Thank you for all the help everyone provided.
